### PR TITLE
Improve enclosures

### LINKED ISSUE #92: Make enclosures more even
## Details:
Hours Spent: 0

## Acceptance Criteria
- [x] Enclosures have an even thickness approximately corresponding to the specified accuracy.
- [x] Where a box enclosure is better than a parallelogram, use the box enclosure.  
- [x] Plot of sqrt(x) should be enclosed by a box near 0.

## Closing Notes

Overview of changes can be found in PR _PR ID HERE_

### PR DESCRIPTION
**Pull request by michalkonecny**
# Issues
Closes #92

# Summary
- Enclosure polygons are either boxes, parallelograms or intersection of both, whichever is best.
- Segmentation now checks for stability of second derivative before stopping bisection.

### CODE REVIEW

File: src/IntervalArith/Approx/Approx.purs [hunk 1]
```diff
@@ -282,6 +282,12 @@ exact (Approx _ _ e _) = e == (big 0)
 
 exact Bottom = false
 
+-- |Returns 'True' if the approximation is finite, i.e., it is not Bottom
+finite :: Approx -> Boolean
```
Comment (RelativeForce) [inline src/IntervalArith/Approx/Approx.purs]: Rename to `isFinite` as it is a boolean check

File: src/Plot/RobustPlot.purs [hunk 2]
```diff
@@ -43,29 +45,109 @@ plotEnclosures canvasSize bounds domainSegments evaluator = segmentEnclosures
...
+            xRight <- evaluator.f xUA
+            Just (Tuple (lowerA xLeft) (upperA xRight))
+          | xGradUpper !<=! zero -> do
+            xLeft <- evaluator.f xLA
+            xRight <- evaluator.f xUA
+            Just (Tuple (lowerA xRight) (upperA xLeft))
+          | otherwise -> map boundsA $ evaluator.f x
+        _ -> map boundsA $ evaluator.f x
+    in
+      -- TODO: computer yLower yUpper by endpoints if gradient is positive or negative
```
Comment (RelativeForce) [inline src/Plot/RobustPlot.purs]: Is this TODO complete?

File: src/Plot/Segments.purs [hunk 3]
```diff
@@ -47,8 +47,21 @@ segmentDomain accuracyTarget evaluator l u = fromFoldable $ segementDomainF 0 l
...
+        "a1 = "
+          <> show a1
+          <> "; a2 = "
+          <> show a2
+          <> "; h = "
+          <> show h
+          <> "; w = "
+          <> show w
+          <> "; accuracyTarget = "
+          <> show accuracyTarget
```
Comment (RelativeForce) [inline src/Plot/Segments.purs]: Remove zombie code as it is unused

**Review by RelativeForce — COMMENTED** [review]
Looks good just some minor adjustments

File: src/IntervalArith/Approx/Approx.purs (hunk 1 above)
Comment (michalkonecny) [reply src/IntervalArith/Approx/Approx.purs]: Thank you.
Renamed finite-> isFinite and exact -> isExact

File: src/Plot/RobustPlot.purs (hunk 2 above)
Comment (michalkonecny) [reply src/Plot/RobustPlot.purs]: Yes, I have just removed the comment.  Thank you.

File: src/Plot/Segments.purs (hunk 3 above)
Comment (michalkonecny) [reply src/Plot/Segments.purs]: Removed.  Thank you.

**Review by RelativeForce — APPROVED** [review]
Looks good

**Pull request merged**